### PR TITLE
Player mind fixes

### PR DESCRIPTION
**Pull request by Bod9001**
so, the problem is 
owning an object =/= Controlling that object

### Changelog:
CL: [Fix] Fixed effects on body being kept When ghosting e.g blurry vision
CL: [Fix] Fixed issues like cloning yourself and then reattaching the head of your ex body forcing you into that body
CL: [Fix] Hopefully fixes admin UI breaking

### CODE REVIEW

**Comment by corp-0** [comment]
> CL: [Fix] fixes some issues with mind rework

what issues

**Comment by corp-0** [comment]
can you elaborate on what is the difference between EnterPlayer and ControlPlayer?

**Comment by Bod9001** [comment]
EnterPlayer  = The player is now controlling show the UI for the Player
ControlPlayer = The player now has possession of this, Not necessarily controlling this moment

**Comment by corp-0** [comment]
> ControlPlayer = The player now has possession of this, Not necessarily controlling this moment

ControlPlayer doesn't necessarilly means controlling?

**Comment by Bod9001** [comment]
yeah, Couldn't really think of better names

**Comment by CorruptComputer** [comment]
What about:
- ControlPlayer = The player is now controlling show the UI for the Player
- PossessPlayer  = The player now has possession of this, Not necessarily controlling this moment

I personally think these names are easier to understand what they are for.

**Comment by corp-0** [comment]
I like that

**Comment by corp-0** [comment]
Fixed your changelog. You have to make sure CL: is the beginning of the line, no spaces before allowed

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 1]
```diff
@@ -44,6 +60,19 @@ public interface IOnPlayerLeaveBody
 	public void OnPlayerLeaveBody(PlayerInfo account);
 }
 
+/// <summary>
+/// This interface will be called when a player is transferred out of their body
+/// </summary>
+public interface IOnPlayerLosePossess
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:66]:
```suggestion
public interface IOnPlayerLosePossession
```

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 2]
```diff
@@ -44,6 +60,19 @@ public interface IOnPlayerLeaveBody
...
+{
+	/// <summary>
+	/// Called on server when the player leaves a body (interface called on the old player object)
+	/// </summary>
+	/// <param name="account">The mind of the player leaving the body</param>
+	public void OnPlayerLosePossess(Mind Mind);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs]:
```suggestion
	public void OnPlayerLosePossession(Mind mind);
```

File: UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs [hunk 3]
```diff
@@ -44,6 +60,19 @@ public interface IOnPlayerLeaveBody
...
+}
+
+
 
 public interface IClientPlayerLeaveBody
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:77]: so what's the difference between losing possession and leaving the body?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:77]: renamed

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs:77]: leaving the body is  OnPlayerLeaveBody, 
losing possession this is when you remove access to view slots / that type of thing

nothing uses it yet though

**Review by corp-0 — APPROVED** [review]

**Pull request merged**